dbt_style_guide.md

```diff
@@ -1,237 +1,520 @@
-```
-├── dbt_project.yml
-└── models
-    ├── marts
-    |   └── core
-    |       ├── intermediate
-    |       |   ├── intermediate.yml
-    |       |   ├── customers__unioned.sql
-    |       |   ├── customers__grouped.sql
-    |       └── core.yml
-    |       └── core.docs
-    |       └── dim_customers.sql
-    |       └── fct_orders.sql
-    └── staging
-        └── stripe
-            ├── base
-            |   ├── base__stripe_invoices.sql
-            ├── src_stripe.yml
-            ├── src_stripe.docs
-            ├── stg_stripe.yml
-            ├── stg_stripe__customers.sql
-            └── stg_stripe__invoices.sql
-```
-- All objects should be plural, such as: `stg_stripe__invoices`
-- Base tables are prefixed with `base__`, such as: `base__<source>_<object>`
-- Intermediate tables should end with a past tense verb indicating the action performed on the object, such as: `customers__unioned`
-- Marts are categorized between fact (immutable, verbs) and dimensions (mutable, nouns) with a prefix that indicates either, such as: `fct_orders` or `dim_customers`
-
-## Model configuration
-
-- Model-specific attributes (like sort/dist keys) should be specified in the model.
-- If a particular configuration applies to all models in a directory, it should be specified in the `dbt_project.yml` file.
-- In-model configurations should be specified like this:
-
-```python
-{{
-  config(
-    materialized = 'table',
-    sort = 'id',
-    dist = 'id'
-  )
-}}
-```
-- Marts should always be configured as tables
 
-## dbt conventions
-* Only `stg_` models (or `base_` models if your project requires them) should select from `source`s.
-* All other models should only select from other models.
+- All objects should be plural.  
+  Example: `stg_stripe__invoices.sql` vs. `stg_stripe__invoice.sql`
 
-## Testing
+- All objects should have a prefix to indicate their DAG stage in the flow.  
+  See [dbt Conventions](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md#dbt-conventions) for more information.
 
-- Every subdirectory should contain a `.yml` file, in which each model in the subdirectory is tested. For staging folders, the naming structure should be `src_sourcename.yml`. For other folders, the structure should be `foldername.yml` (example `core.yml`).
-- At a minimum, unique and not_null tests should be applied to the primary key of each model.
+- All models should use the naming convention `<type/dag_stage>_<source/topic>__<additional_context>`. See [this article](https://docs.getdbt.com/blog/stakeholder-friendly-model-names) for more information.
+  - For models in the **marts** folder `__<additional_context>` is optional. 
+  - Models in the **staging** folder should use the source's name as the `<source/topic>` and the entity name as the `additional_context`.
 
-## Naming and field conventions
+    Examples:
+    - seed_snowflake_spend.csv
+    - base_stripe__invoices.sql
+    - stg_stripe__customers.sql

```

Comment:
It might be helpful to add the example stg_salesforce__customers.sql to demonstrate the need for including the source in the name of every stg model - even when it gets long.  
```suggestion
    - stg_stripe__customers.sql
    - stg_salesforce__customers.sql
```